app/assets/javascripts/admin/addon/components/site-setting.gjs

```diff
@@ -50,13 +53,48 @@ export default class SiteSettingComponent extends Component {
   @service router;
   @service dialog;
   @service siteSettingChangeTracker;
+  @service messageBus;
 
   @tracked isSecret = null;
+  @tracked status = null;
+  @tracked progress = null;
   updateExistingUsers = null;
 
   constructor() {
     super(...arguments);
     this.isSecret = this.setting?.secret;
+
+    const settingName = this.setting.setting;
+
+    if (
+      !!settingName &&
+      (settingName.includes("default_categories") ||
+        settingName.includes("default_tags"))
```

Comment:
The logic here can be extracted in a function so that it is not duplicated across the `constructor()` and `willDestroy()` function.